cub/util_type.cuh

```diff
@@ -273,9 +274,47 @@ struct NullType
 template <int A>
 struct Int2Type
 {
-   enum {VALUE = A};
+    enum {VALUE = A};
 };
 
+/**
+ * \brief Allows algorithms that take a value as input to take a future value that is not computed yet at launch time.
+ *
+ * Note that it is user's responsibility to ensure that the result will be ready before use via external synchronization
+ * or stream-ordering dependencies.
+ */
+template <typename T, typename IterT = T*>
+struct FutureValue
+{
+    using value_type = T;
+    using iterator_type = IterT;
+    IterT iter;
+    __host__ __device__ __forceinline__ FutureValue(IterT iter):iter(iter) {}
```

Comment:
- This constructor should be marked `explicit`.
- Use a different name than "iter" for either the argument or member to avoid shadowing warnings. I'd suggest renaming the member variable `m_iter`.
- Make the internal IterT member variable private, since users shouldn't directly interact with it.

Reply:
fixed